src/database.cpp

```diff
@@ -208,6 +208,16 @@ void Database::removeLoc(QString locname) {
         qDebug() << "ERROR: " << query.lastError().text();
 }
 
+int Database::AvaliableSeat(QString BusName, QString Origin, QString dest, QString date, QString time) {
+    qDebug() << "CHECKING AVALIABLE SEATS ";
+    QSqlQuery query;
+    query.prepare("SELECT COUNT(SEAT_NO) FROM TICKET WHERE BUSNAME = ? AND ORIGIN = ? AND DEST = ?AND DATE = ? AND TIME = ?");
+    int TakenSeat = query.value(0).toInt();
+    int availableSeat=30-TakenSeat;
+
+    if (!query.exec())
+        qDebug() << query.lastError().text();
+
```

Comment:
Oh, you forgot to close the curly brace `}` in `AvailableSeat` func, please don't make this kind of silly mistakes. I know you can't run the code in PC, but at least you should check for typing mistakes before comitting